packages/eas-cli/src/commands/channel/rollout.ts

```diff
@@ -352,12 +338,18 @@ export default class ChannelRollout extends EasCommand {
     const fullName = await getProjectFullNameAsync(exp);
     const projectId = await getProjectIdAsync(exp);
 
-    const getUpdateChannelByNameForAppResult = await getUpdateChannelByNameForAppAsync({
+    const channel = await ChannelQuery.getUpdateChannelByNameForAppAsync({
       appId: projectId,
       channelName: channelName!,
     });
+    if (!channel) {
+      throw new Error(
+        `Could not find a channel named "${channelName}". Please check what channels exist on this project with "eas channel:list".`
+      );
+    }
+
     const { branchMapping: currentBranchMapping, isRollout } = getBranchMapping(
-      getUpdateChannelByNameForAppResult.app?.byId.updateChannelByName?.branchMapping
+      channel.branchMapping
```

Comment:
previously this would throw with an opaque error about undefined not having a branchMapping attribute